manager/controlapi/service.go

```diff
@@ -839,8 +839,28 @@ func (s *Server) RemoveService(ctx context.Context, request *api.RemoveServiceRe
 		return nil, status.Errorf(codes.InvalidArgument, errInvalidArgument.Error())
 	}
 
+	var service *api.Service
 	err := s.store.Update(func(tx store.Tx) error {
-		return store.DeleteService(tx, request.ServiceID)
+		service = store.GetService(tx, request.ServiceID)
+		if service == nil {
+			return status.Errorf(codes.NotFound, "service %s not found", request.ServiceID)
+		}
+
+		// if the service has no running tasks, we can remove it right away
+		m := service.Spec.GetMode()
+		switch m.(type) {
+		case *api.ServiceSpec_Replicated:
+			if int64(m.(*api.ServiceSpec_Replicated).Replicated.Replicas) == 0 {
+				return store.DeleteService(tx, request.ServiceID)
+			}
+		case *api.ServiceSpec_Global:
```

Comment:
@anshulpundir @dperny suggestions for whether or not we want to handle this case here. Not sure what the right condition would be here, but for checking nodes in the cluster or something like that.

Reply:
Can we just get tasks by service ?

Reply:
Sorry for the long silence on this PR. What happens if say the cluster is a small number of nodes, and all tasks restart around the same time for some reason? This is probably a corner case that may not be critical. The issue here is that the service spec for global services isn't enough to make this decision unlike replicated services.

Reply:
Sorry, I'm not sure I follow. Lets talk in person.